### ISSUE TITLE
Add iscsi as secondary storage

### ISSUE THREAD

**Issue by aleskxyz** [apache/cloudstack#3751]
It seems iscsi has better performance in comparison to NFS, specially in vmware environment.
So I think it will be a good idea to add support of iscsi to secondary storage.

**Comment by svenvogel** [comment]
@geekza1 that would be nice but i think implementation is not easy. since we use managed storage we would recommend it.

**Comment by geekza1** [comment]
> @geekza1 that would be nice but i think implementation is not easy. since we use managed storage we would recommend it.

Could I know what is its name?
Is it compatible with vmware?
If it uses nfs to connect to vmware we may should have the same problem!

**Comment by bwsw** [comment]
Why one needs that if one can use Ceph RBD or S3 (Minio) or even Gluster with minor hacks? I don't think block device is an easy way to use as a SS, because it's necessary to add certain kind of addressation inside (actual filesystem) and DLM.

**Comment by DaanHoogland** [comment]
@aleskxyz there are improved storage types support for vmware, I don't think for secondary but you might want to look at those (and maybe use direct download if you don't want an nfs store).
@svenvogel is referring to solidfire i think !?!

**Comment by aleskxyz** [comment]
@bwsw The downside of all the mentioned solutions is vmware doesn't mount them directly. An NFS staging server is required for them.
@DaanHoogland Cloudstack only supports certain type of SS that all of them needs an NFS server. I don't get your point about "direct download". Did you mean that I can use an vmfs storage as my SS by editing entries in database?

**Comment by DaanHoogland** [comment]
@aleskxyz please review the release candidate for 4.15. It does support different types of storage on vmware.

**Comment by DaanHoogland** [comment]
> @DaanHoogland Cloudstack only supports certain type of SS that all of them needs an NFS server. I don't get your point about "direct download". Did you mean that I can use an vmfs storage as my SS by editing entries in database?

@aleskxyz `direct download` means that templates are not downloaded and stored but the remote url is registered and whenever a deploy occurs the template is downloaded to primary if it is not yet available on it. This seems to partially fit your use-case (I don't know if you use backup of snapshots or the likes)

**Comment by DaanHoogland** [comment]
@winterhazel are you planning any implementation for this?

**Comment by winterhazel** [comment]
@DaanHoogland yes. I am experimenting some approaches to add support for new secondary storage interfaces, iSCSI included.
